File: frontend/src/types/asset.ts
export type AssetType = 'SAVINGS' | 'CREDIT_CARD' | 'CASH' | 'ETC'

export const ASSET_TYPE_LABELS: Record<AssetType, string> = {
  SAVINGS: '예금/적금',
  CREDIT_CARD: '신용카드',
  CASH: '현금',
  ETC: '기타',
}

export interface Asset {
  id: number
  name: string
  initialAmount: number
  currentBalance: number  // initialAmount + 연결된 거래 합계 (백엔드 자동 계산)
  assetType: AssetType
  createdAt: string
}

export interface AssetRequest {
  name: string
  initialAmount: number
  assetType: AssetType
}

File: backend/src/main/java/com/cashbook/domain/asset/entity/Asset.java
package com.cashbook.domain.asset.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Asset {
    private Long id;
    private Long userId;
    private String name;
    private Long initialAmount;
    private String assetType;   // SAVINGS, CREDIT_CARD, CASH, ETC
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
    private OffsetDateTime deletedAt;
}
